sql/join.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+package sql
+
+import (
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+const joinBatchSize = 100
+
+// An indexJoinNode implements joining of results from an index with the rows
+// of a table. The index side of the join is pulled first and the resulting
+// rows are used to lookup rows in the table. The work is batched: we pull
+// joinBatchSize rows from the index and use the primary key to construct spans
+// that are looked up in the table.
+type indexJoinNode struct {
+	index            *scanNode
+	table            *scanNode
```

Comment:
is there value to embedding this so that you don't have to implement Columns(), Values() and Ordering()?